### PR TITLE
switch to GET method when taking user consent

### LINKED ISSUE #35: nonce missing in id_token at first SSO attempt
First SSO attempt of a new user first attempt fails and Synapse complains about nonce being missing from id_token but then second attempt passes as its included.

### PR DESCRIPTION
**Pull request by ashfame**
Currently bshaffer oauth library has a bug when POST is used on AuthorizeEndpoint along with nonce (optional parameter) which fails to set the nonce in id_token

This PR switches the form submit method to `GET` when taking user consent and this fixes the issue.

Fixes #35 

The reason why this issue would only break for new users is because of sticky consent functionality the parameters would get passed as GET parameters, hence not uncovering the issue with bshaffer oauth library.

Additionally, I have also already created a [PR](https://github.com/bshaffer/oauth2-server-php/pull/1032) to the library with the fix.

### CODE REVIEW

File: src/Http/Handlers/AuthorizeHandler.php [hunk 1]
```diff
@@ -36,6 +36,7 @@ public function handle( Request $request, Response $response ): Response {
 
 		$user = wp_get_current_user();
 		if ( $this->consent_storage->needs_consent( $user->ID ) ) {
+			// phpcs:disable WordPress.Security.NonceVerification.Recommended We already have REST API nonce validating the user intention with this request
```
Comment (psrpinto) [inline src/Http/Handlers/AuthorizeHandler.php]: I think `phpcs:disable` applies to the whole file, you could use `phpcs:ignore` to ignore the next line.

File: templates/authenticate/form.php [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <?php /** @var stdClass $data */ ?>
 
-<form method="post" action="<?php echo esc_url( $data->form_url ); ?>">
+<form method="GET" action="<?php echo esc_url( $data->form_url ); ?>">
```
Comment (psrpinto) [inline templates/authenticate/form.php:3]: Would there be a way to not have to change this to `GET`, maybe by transforming the request on the server side from POST to GET (before the call to `handleAuthorizeRequest`)?

I'm asking this because `form.php` is part of our "public API", in the sense that users might have overridden this template, so they wouldn't get this change. I realise no one is probably using this library yet, but I think the principle still applies.

In short we're changing our "public API" because of a bug in a library we're using internally. I think It would be best if the template would not be dependent on the fact we're using the bshaffer library, as we might stop using it in the future, or the bug might get fixed.

**Review by psrpinto — COMMENTED** [review]
Thanks for also opening a PR to the library!

File: templates/authenticate/form.php (hunk 2 above)
Comment (ashfame) [reply templates/authenticate/form.php]: I understand your concern and based on Alex's suggestion I have copied over the value with explanation in the comment.

**Comment by ashfame** [comment]
@psrpinto @akirk Please take another look! Thanks!

**Review by psrpinto — APPROVED** [review]
👍

**Comment by ashfame** [comment]
@akirk Going to merge this but happy to address any remarks, if there are any, later on.
